hyppo/ksample/smoothCF.py

```diff
@@ -0,0 +1,155 @@
+import numpy as np
+from numba import jit
+from scipy.stats import chi2
+from ._utils import _CheckInputs
+from hyppo.ksample.base import KSampleTest, KSampleTestOutput
+
+
+class SmoothCFTest(KSampleTest):
+    r"""
+    Smooth Characteristic Function test statistic and p-value
+
+    The Smooth Characteristic Function test is a two-sample test that uses
+    differences in the smoothed (analytic) characteristic function of
+    two data distributions in order to determine how different
+    the two data are.
+
+    Notes
+    ____
+
+    The test statistic takes on the following form:
+
+    :math: 'nW_n\Sigma_n^{-1}W_n'
+
+    As seen in the above formulation, this test-statistic takes the same form as
+    the Hotelling :math: 'T^2' statistic. However, the components are
+    defined differently in this case. Given data sets
+    X and Y, define the following as :math: 'Z_i', the vector of differences:
+
+    :math: 'Z_i = (k(X_i, T_1) - k(Y_i, T_1), \ldots, k(X_i, T_J) - k(Y_i, T_J)) \in mathbb{R}^J'
+
+    The above is the vector of differences between kernels at test points, :math: 'T_j'.
+    This same formulation is used in the Mean Embedding Test.
+    Moving forward, :math: 'W_n' can be defined:
+
+    :math: 'W_n = \frac{1}{n} \sum_{i = 1}^n Z_i
+
+    This leaves :math: '\Sigma_n', the covariance matrix as:
+
+    :math: '\Sigma_n = \frac{1}{n}ZZ^T'
+
+    In the specific case of the Smooth Characteristic function test,
+    the vector of differences can be defined as follows:
+
+    :math: 'Z_i = (f(X_i)\sin(X_iT_1) - f(Y_i)\sin(Y_iT_1), f(X_i)\cos(X_iT_1) - f(Y_i)\cos(Y_iT_1),
+    \cdots) \in \mathbb{R}^{2J}'
+
+    Once :math: 'S_n' is calculated, a threshold :math: 'r_{\alpha}' corresponding to the
+    :math: '1 - \alpha' quantil of a Chi-squared distribution w/ J degrees of freedom
+    is chosen. Null is rejected if :math: 'S_n' is larger than this threshold.
+
+    References
+    ----------
+    .. footbibliography::
+    """
+    def __init__(self, scale=1.0, num_random_features=5, **kwargs):
+
+        self.scale = scale
+        self.num_random_features = num_random_features
+        KSampleTest.__init__(
+            self,  **kwargs
+        )
+
+    def statistic(self, x, y):
+        """
+        :return: test statistic for smoothCF
```

Comment:
no Parameters or inputs section. Use numpydoc format as in the rest of hyppo

Reply:
changed

Reply:
changed